sky/jobs/server/core.py

```diff
@@ -186,6 +242,190 @@ def _maybe_submit_job_locally(prefix: str, dag: 'sky.Dag',
+        remote_orig_user_yaml_path = (
+            f'{prefix}/ensure-controller-up-{dag_uuid}.original_user_yaml')
+        remote_user_yaml_path = f'{prefix}/ensure-controller-up-{dag_uuid}.yaml'
+        remote_user_config_path = (
+            f'{prefix}/ensure-controller-up-{dag_uuid}.config_yaml')
+        remote_env_file_path = f'{prefix}/ensure-controller-up-{dag_uuid}.env'
+
+        # Create minimal temporary files for template
+        with tempfile.NamedTemporaryFile(
+                prefix='ensure-controller-up-', mode='w',
+                delete=False) as f, tempfile.NamedTemporaryFile(
+                    prefix='ensure-controller-up-user-', mode='w',
+                    delete=False) as original_user_yaml_path:
+            # Write minimal dag content
+            minimal_dag = task_lib.Task(name='ensure_controller_up',
+                                        run='echo "Controller cluster is up"')
+            with dag_lib.Dag() as temp_dag:
+                temp_dag.add(minimal_dag)
+                dag_utils.dump_chain_dag_to_yaml(temp_dag, f.name)
+            original_user_yaml_path.write('')
+            original_user_yaml_path.flush()
+
+            vars_to_fill: Dict[str, Any] = {
+                'remote_original_user_yaml_path': remote_orig_user_yaml_path,
+                'original_user_dag_path': original_user_yaml_path.name,
+                'remote_user_yaml_path': remote_user_yaml_path,
+                'user_yaml_path': f.name,
+                'local_to_controller_file_mounts': {},
+                'jobs_controller': controller_name,
+                'dag_name': 'ensure_controller_up',
+                'remote_user_config_path': remote_user_config_path,
+                'remote_env_file_path': remote_env_file_path,
+                'modified_catalogs': {},
+                'priority': skylet_constants.DEFAULT_PRIORITY,
+                'is_consolidation_mode': False,
+                'pool': None,
+                'job_controller_indicator_file':
+                    managed_job_constants.JOB_CONTROLLER_INDICATOR_FILE,
+                **controller_utils.shared_controller_vars_to_fill(
+                    controller,
+                    remote_user_config_path=remote_user_config_path,
+                    local_user_config={},
+                ),
+            }
+
+            yaml_path = os.path.join(
+                managed_job_constants.JOBS_CONTROLLER_YAML_PREFIX,
+                f'ensure-controller-up-{dag_uuid}.yaml')
+
+            # Fill the template to create the controller task YAML
+            common_utils.fill_template(
+                managed_job_constants.JOBS_CONTROLLER_TEMPLATE,
+                vars_to_fill,
+                output_path=yaml_path)
+
+            # Create task from the template-generated YAML
+            controller_task = task_lib.Task.from_yaml(yaml_path)
+            controller_task.set_resources(controller_resources_set)
```

Comment:
Rather than doing all this complicated stuff, can we just fork the controller yaml into one yaml for setting up the cluster and another yaml for submitting jobs?